mongo/integration/client_test.go

```diff
@@ -768,6 +768,88 @@ func TestClient(t *testing.T) {
 				"expected 'OP_MSG' OpCode in wire message, got %q", pair.Sent.OpCode.String())
 		}
 	})
+
+	opts := mtest.NewOptions().
+		// Blocking failpoints don't work on pre-4.2 and sharded clusters.
+		Topologies(mtest.Single, mtest.ReplicaSet).
+		MinServerVersion("4.2").
+		// Expliticly enable retryable reads and retryable writes.
+		ClientOptions(options.Client().SetRetryReads(true).SetRetryWrites(true))
+	mt.RunOpts("operations don't retry after a context timeout", opts, func(mt *mtest.T) {
+		testCases := []struct {
+			desc      string
+			failPoint mtest.FailPoint
+			operation func(context.Context, *mongo.Collection) error
+		}{
+			{
+				desc: "read op",
+				failPoint: mtest.FailPoint{
+					ConfigureFailPoint: "failCommand",
+					Mode:               "alwaysOn",
+					Data: mtest.FailPointData{
+						FailCommands:    []string{"find"},
+						BlockConnection: true,
+						BlockTimeMS:     500,
+					},
+				},
+				operation: func(ctx context.Context, coll *mongo.Collection) error {
+					return coll.FindOne(ctx, bson.D{}).Err()
+				},
+			},
+			{
+				desc: "write op",
+				failPoint: mtest.FailPoint{
+					ConfigureFailPoint: "failCommand",
+					Mode:               "alwaysOn",
+					Data: mtest.FailPointData{
+						FailCommands:    []string{"insert"},
+						BlockConnection: true,
+						BlockTimeMS:     500,
+					},
+				},
+				operation: func(ctx context.Context, coll *mongo.Collection) error {
+					_, err := coll.InsertOne(ctx, bson.D{})
+					return err
+				},
+			},
+		}
```

Comment:
What are your thoughts of taking the failPoint out of the test case? Since the test runner assumes that the failpoint is 500 ms, it's not  really a configurable datapoint. The only difference between the different failpoints is the `FailCommand`:

```go
failPoint := mtest.FailPoint{
	ConfigureFailPoint: "failCommand",
	Mode:               "alwaysOn",
	Data: mtest.FailPointData{
		FailCommands:    []string{"find", "insert"},
		BlockConnection: true,
		BlockTimeMS:     500,
	},
}

mt.SetFailPoint(failPoint)

testCases := []struct {
	desc      string
	operation func(context.Context, *mongo.Collection) error
}{
	{
		desc: "read op",
		operation: func(ctx context.Context, coll *mongo.Collection) error {
			return coll.FindOne(ctx, bson.D{}).Err()
		},
	},
	{
		desc: "write op",
		operation: func(ctx context.Context, coll *mongo.Collection) error {
			_, err := coll.InsertOne(ctx, bson.D{})
			return err
		},
	},
}
```

Reply:
Makes sense, I'll update it.